### PR TITLE
Refactor openevec to base actions on top of type and fix multiple configs

### PR DESCRIPTION
**Pull request by giggsoff**
To use the same semantic in openevec we should use the same base type.
Also we use novel changers to load config directly to be able to load config directly from openevec.

### CODE REVIEW

**Comment by uncleDecart** [comment]
@giggsoff I see from review that you are removing cfg from parameters and rely on reading config from file which creates unnecessary coupling. That brings back things that I tried to revert with my previous PRs: you don't have single configuration file object, you will have to use `./eden config add` before using any eden command.

I believe that this is not a way to go because you cannot simply create golang structure and use it when calling any openEVEC functions and pass it (see [here](https://github.com/lf-edge/eden/pull/873/files#diff-87691c3e7dc3504ef904df0cbdca8cc6448043c421c479163a6f7080d5e530d1R34))

**Comment by giggsoff** [comment]
> @giggsoff I see from review that you are removing cfg from parameters and rely on reading config from file which creates unnecessary coupling. That brings back things that I tried to revert with my previous PRs: you don't have single configuration file object, you will have to use `./eden config add` before using any eden command.
> 
> I believe that this is not a way to go because you cannot simply create golang structure and use it when calling any openEVEC functions and pass it (see [here](https://github.com/lf-edge/eden/pull/873/files#diff-87691c3e7dc3504ef904df0cbdca8cc6448043c421c479163a6f7080d5e530d1R34))

Probably it is not so clear, but I try to do exactly what you mention. Instead of dirty magic inside of re-reading config from viper [here](https://github.com/lf-edge/eden/blob/0af365aa1dc04434f4548e68dc16e11e9bf7d2cf/pkg/utils/config.go#L68) I fixed and use [InitVarsFromConfig](https://github.com/lf-edge/eden/blob/0af365aa1dc04434f4548e68dc16e11e9bf7d2cf/pkg/openevec/test.go#L30 ) which rely on config object you implemented before (`EdenSetupArgs`).
So, in you example, you will use something like:
```
cfg := ec.GetDefaultConfig(curPath)
openEVEC = openevec.CreateOpenEVEC(cfg)
...
openEVEC.PodDeploy(appLink, *pc)
```

File: cmd/root.go [hunk 1]
```diff
@@ -12,14 +12,15 @@ import (
 	"github.com/spf13/cobra"
 )
 
+var openEVEC *openevec.OpenEVEC
```
Comment (uncleDecart) [inline cmd/root.go:15]: Do you think it makes sense to pass this object during the construction of CLI?

Comment (giggsoff) [reply cmd/root.go:15]: Actually we need `configName` to initialize it properly (with particular config), so I do it in `PersistentPreRunE` where we have it.

File: cmd/edenPacket.go [hunk 2]
```diff
@@ -53,7 +53,7 @@ func newPacketRunCmd(cfg *openevec.EdenSetupArgs, packetKey, packetProjectName *
 		Use:   "run",
 		Short: "run vm in packet",
 		Run: func(cmd *cobra.Command, args []string) {
-			err := openevec.PacketRun(*packetKey, *packetProjectName, packetVMName, packetZone, packetMachineType, packetIPXEUrl, cfg)
```
Comment (uncleDecart) [inline cmd/edenPacket.go:56]: Seems like you don't need cfg in function anymore

File: cmd/edenPod.go [hunk 3]
```diff
@@ -57,7 +57,7 @@ func newPodPublishCmd(cfg *openevec.EdenSetupArgs) *cobra.Command {
 		Args:  cobra.ExactArgs(1),
 		Run: func(cmd *cobra.Command, args []string) {
 			appName := args[0]
-			if err := openevec.PodPublish(appName, kernelFile, initrdFile, rootFile, formatStr, arch, local, disks, cfg); err != nil {
+			if err := openEVEC.PodPublish(appName, kernelFile, initrdFile, rootFile, formatStr, arch, local, disks); err != nil {
```
Comment (uncleDecart) [inline cmd/edenPod.go:60]: seems like no need for cfg in function as well

File: cmd/edenPacket.go (hunk 2 above)
Comment (giggsoff) [reply cmd/edenPacket.go:56]: Sorry, can you please elaborate, what exactly you want to see here?

Comment (uncleDecart) [reply cmd/edenPacket.go:56]: Unfortunately, I can't commit to your PR :D What I meant is yetus is complaining on unused variable, we can fix it by

```
diff --git a/cmd/edenPacket.go b/cmd/edenPacket.go
index 3f0e22e..82c60c1 100644
--- a/cmd/edenPacket.go
+++ b/cmd/edenPacket.go
@@ -16,7 +16,7 @@ func newPacketCmd(configName, verbosity *string) *cobra.Command {
                Short: `Manage VMs in Equinix Metal Platform`,
        }
 
-       packetCmd.AddCommand(newPacketVMCmd(cfg, &cfg.Packet.Key, &packetProjectName))
+       packetCmd.AddCommand(newPacketVMCmd(&cfg.Packet.Key, &packetProjectName))
 
        packetCmd.PersistentFlags().StringVarP(&cfg.Packet.Key, "key", "k", "", "packet key file")
        packetCmd.PersistentFlags().StringVarP(&packetProjectName, "project", "p", defaults.DefaultPacketProjectName, "project name on packet")
@@ -24,7 +24,7 @@ func newPacketCmd(configName, verbosity *string) *cobra.Command {
        return packetCmd
 }
 
-func newPacketVMCmd(cfg *openevec.EdenSetupArgs, packetKey, packetProjectName *string) *cobra.Command {
+func newPacketVMCmd(packetKey, packetProjectName *string) *cobra.Command {
        var packetVMCmd = &cobra.Command{
                Use:   "vm",
                Short: `Manage VMs in packet`,
@@ -34,7 +34,7 @@ func newPacketVMCmd(cfg *openevec.EdenSetupArgs, packetKey, packetProjectName *s
                {
                        Message: "Basic Commands",
                        Commands: []*cobra.Command{
-                               newPacketRunCmd(cfg, packetKey, packetProjectName),
+                               newPacketRunCmd(packetKey, packetProjectName),
                                newPacketDeleteCmd(packetKey, packetProjectName),
                                newPacketGetIPCmd(packetKey, packetProjectName),
                        },
@@ -46,7 +46,7 @@ func newPacketVMCmd(cfg *openevec.EdenSetupArgs, packetKey, packetProjectName *s
        return packetVMCmd
 }
```

**Comment by uncleDecart** [comment]
@giggsoff I'll merge it and will follow up with PR to fix Yetus complains about unused variables

**Review by uncleDecart — APPROVED** [review]

**Comment by uncleDecart** [comment]
follow-up PR https://github.com/lf-edge/eden/pull/907

**Pull request merged**